engine/model_xpubs.go

```diff
@@ -19,8 +19,8 @@ type Xpub struct {
 	// Model specific fields
 	ID              string `json:"id" toml:"id" yaml:"id" gorm:"<-:create;type:char(64);primaryKey;comment:This is the sha256(xpub) hash" bson:"_id"`
 	CurrentBalance  uint64 `json:"current_balance" toml:"current_balance" yaml:"current_balance" gorm:"<-;comment:The current balance of unspent satoshis" bson:"current_balance"`
-	NextInternalNum uint32 `json:"next_internal_num" toml:"next_internal_num" yaml:"next_internal_num" gorm:"<-;type:int;comment:The next index number for the internal xPub derivation" bson:"next_internal_num"`
-	NextExternalNum uint32 `json:"next_external_num" toml:"next_external_num" yaml:"next_external_num" gorm:"<-;type:int;comment:The next index number for the external xPub derivation" bson:"next_external_num"`
+	NextInternalNum uint32 `json:"next_internal_num" toml:"next_internal_num" yaml:"next_internal_num" gorm:"<-;type:int not null;default:0comment:The next index number for the internal xPub derivation" bson:"next_internal_num"`
+	NextExternalNum uint32 `json:"next_external_num" toml:"next_external_num" yaml:"next_external_num" gorm:"<-;type:int not null;default:1;comment:The next index number for the external xPub derivation" bson:"next_external_num"`
```

Comment:
Maybe I'm wrong, or missing some piece,
but I think that having separated `xpub.next_internal_num`, `xpub.next_external_num`, `paymail.external_xpub_key_nume` and `paymail.pub_key_num` can probably lead to some collisions (reusage of the derived keys)

Reply:
Fixed/changed as we discussed it